### ISSUE TITLE
Modifying the value of above_rule_id triggers a replacement of all resources

### ISSUE THREAD

**Issue by GeorgiosGoniotakis** [vmware/terraform-provider-vcd#624]
### Details
Terraform Version: 0.14.4

### Affected Resource(s)
Please list the resources as a list, for example:
- vcd_nsxv_firewall_rule

### Description of Issue
Hello there and thank you for your contribution to open source. We are currently using the property `above_rule_id` mentioned on Example no 4 of the docs for `vcd_nsxv_firewall_rule`. 

```hcl
resource "vcd_nsxv_firewall_rule" "my-rule-1" {
  org          = "my-org"
  vdc          = "my-vdc"
  edge_gateway = "my-edge-gateway"

  logging_enabled = "true"
  action          = "deny"

  source {
    ip_addresses = ["30.10.10.0/24", "31.10.10.0/24"]
    org_networks = ["org-net-1", "org-net-2"]
  }

  destination {
    ip_addresses = ["any"]
  }

  service {
    protocol = "icmp"
  }
}

resource "vcd_nsxv_firewall_rule" "my-rule-2" {
  org          = "my-org"
  vdc          = "my-vdc"
  edge_gateway = "my-edge-gateway"

  # This attribute allows to ensure rule is inserted above the referred one
  # in rule processing engine
  above_rule_id = "${vcd_nsxv_firewall_rule.my-rule-1.id}"
  name          = "my-friendly-name"

  source {
    ip_addresses = ["30.10.10.0/24", "31.10.10.0/24"]
    org_networks = ["org-net-1", "org-net-2"]
  }

  destination {
    ip_addresses = ["any"]
  }

  service {
    protocol = "icmp"
  }
}
```

We have created a chain of about 30 rules which use the `above_rule_id` property to create a prioritized list. We are facing the issue that if we need to remove a resource (rule) all of the resources have to be replaced (destroyed and re-created). What would be a way to add or remove a firewall rule without having to re-create all the rules of the set? At this stage we are still experimenting so we could potentially restructure our Terraform code in order to fulfil this requirement. Any guidance would be much appreciated.

**Comment by Didainius** [comment]
Hello @GeorgiosGoniotakis ,
Thanks for the issue. I am afraid we are limited by API capabilities here therefore changing `above_rule_id` requires to recreate the rule.

Is it a scenario that occurs often? One other thought/option would be to chain by using `depends_on` builtin. That way Terraform would not trigger firewall rule creation in parallel and you would have it ordered.

**Comment by GeorgiosGoniotakis** [comment]
Hello @Didainius Thank you for your response. I am afraid that `depends_on` is not going to work quite well in this case. On first deployment it's going to work as expected. However, if we wanted to add an additional rule in the process `depends_on` is not going to have an effect on past resources since they will have been deployed already. As a result Terraform will build a plan which includes the addition of just the new rule. I noticed that in previous versions of the provider there was a resource which was allowing the creation of several firewall rules as a single resource [example here](https://i1.wp.com/anthonyspiteri.net/wp-content/uploads/2019/04/vCD_Terraform_API_3.png?ssl=1). Is there a specific reason why that resource has been deprecated? I can see that in version 34 of the API this operation is still supported.

**Comment by Didainius** [comment]
Yes,
These tags don't seem to be working properly. I am thinking further. Could [`ignore_changes`](https://www.terraform.io/docs/configuration/meta-arguments/lifecycle.html#ignore_changes) help? 
I remember `above_rule_id` was the only mechanism for rule positioning in API. The API is more fit for UI in this case as UI applies all rules at once (see below why we split this into 1entity:1HCL definition)

As for the older deprecated resource - it used internal API which had other problems and is not recommended by VCD anymore. This resource should be superior in terms of it capabilities, but some API limitations like that cause issues. (there may be cases when some metadata is cleaned up on updates of this legacy API)

The reason we have split it is because this is more "native" Terraform approach (1 entity:1 HCL definition) as Terraform natively has this `.id` field for entity ID. Additionally - because this is a much more sophisticated resource (has deeper nested configuration options) Terraform SDK is not fit for that purpose very well (deep nesting of definitions) and could problems when doing `plan` (report changes in all rules, when only one was changed for example)

**Comment by GeorgiosGoniotakis** [comment]
Thank you for the response @Didainius I really appreciate. We will be proceeding with finding a solution internally.

**Comment by Didainius** [comment]
Thanks @GeorgiosGoniotakis ,
Indeed it would be interesting if you can share your ideas once you come up with a way. We are going to work on NSX-T firewall (#622) and I would appreciate if you could share your ideas. There may be as well areas to improve how we do resources so input is always valuable.
